src/fondant/executor.py

```diff
@@ -123,7 +123,8 @@ def _add_and_parse_args(cls, spec: ComponentSpec):
                 help=arg.description,
             )
 
-        return parser.parse_args()
+        args, _ = parser.parse_known_args()
```

Comment:
this is needed to avoid parsing the `ref` when running `fondant execute main` and only parse kubeflow arguments